src/main/java/com/microsoft/sqlserver/jdbc/KeyVaultTokenCredential.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package com.microsoft.sqlserver.jdbc;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.util.logging.ClientLogger;
+import com.microsoft.aad.msal4j.ClientCredentialFactory;
+import com.microsoft.aad.msal4j.ClientCredentialParameters;
+import com.microsoft.aad.msal4j.ConfidentialClientApplication;
+import com.microsoft.aad.msal4j.IAuthenticationResult;
+import com.microsoft.aad.msal4j.IClientCredential;
+import com.microsoft.aad.msal4j.SilentParameters;
+import java.net.MalformedURLException;
+import java.text.MessageFormat;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.HashSet;
+import java.util.concurrent.CompletableFuture;
+import reactor.core.publisher.Mono;
+
+
+/**
+ * An AAD credential that acquires a token with a client secret for an AAD application.
+ */
+@Immutable
+class KeyVaultTokenCredential implements TokenCredential {
+    private final ClientLogger logger = new ClientLogger(KeyVaultTokenCredential.class);
```

Comment:
For consistency, please change this to use the driver's logger.  You can take a look at KerbAuthentication.java for reference usage.